### ISSUE TITLE
JTNN example -- floating point exception

### ISSUE THREAD

**Issue by pschwllr** [dmlc/dgl#370]
## 🐛 Bug

I get a floating point exception on `loss.backward()`, when executing `dgl/examples/pytorch/jtnn/vaetrain_dgl.py`. 

I was wondering if this was also observed by others. 

```
dgl/examples/pytorch/jtnn/vaetrain_dgl.py(102)train()
loss.backward()
(Pdb++) n
[1]    36571 floating point exception  NOCUDA=1 python vaetrain_dgl.py
```
## To Reproduce

1. NOCUDA=1 python vaetrain_dgl.py

## Environment

 - DGL Version (e.g., 1.0): dgl 0.1.3 <pip>
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3):  PyTorch 1.0.0
 - OS (e.g., Linux): Darwin-18.2.0-x86_64-i386-64bit
 - How you installed DGL (`conda`, `pip`, source): pip
 - Python version: 3.5.6
 - CUDA/cuDNN version (if applicable): ---
 - GPU models and configuration (e.g. V100): ---

**Comment by BarclayII** [comment]
Seems that PyTorch changed the behavior of empty tensors quite a lot in 1.0 and it has a bug (?) in gradient computation of an empty boolean-masked slicing of an array.  I filed an issue [there](https://github.com/pytorch/pytorch/issues/16635).

**Comment by BarclayII** [comment]
Fixed in #371.
This is only related to PyTorch 1.0, so it should also work with current release (0.1.3).
